adapters/ix/ix_test.go

```diff
@@ -263,51 +348,72 @@ func TestIxNoContent(t *testing.T) {
 func TestIxInvalidCallMissingSize(t *testing.T) {
 
 	server := httptest.NewServer(
-		http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			resp := openrtb.BidResponse{}
-			js, err := json.Marshal(resp)
-			if err != nil {
-				http.Error(w, err.Error(), http.StatusInternalServerError)
-				return
-			}
-			w.Header().Set("Content-Type", "application/json")
-			w.Write(js)
-		}),
+		http.HandlerFunc(dummyIXServer),
 	)
 	defer server.Close()
 
 	conf := *adapters.DefaultHTTPAdapterConfig
 	an := NewIxAdapter(&conf, server.URL)
 	ctx := context.TODO()
 	pbReq := pbs.PBSRequest{}
+	adUnit := getAdUnit()
+	adUnit.Sizes = []openrtb.Format{}
 	pbBidder := pbs.PBSBidder{
 		BidderCode: "bannerCode",
 		AdUnits: []pbs.PBSAdUnit{
-			{
-				Code:       "unitCode",
-				MediaTypes: []pbs.MediaType{pbs.MEDIA_TYPE_BANNER},
-				BidID:      "bidid",
-				Params:     json.RawMessage("{\"siteId\": \"12\"}"),
-			},
+			adUnit,
 		},
 	}
-	_, err := an.Call(ctx, &pbReq, &pbBidder)
-	if err == nil {
+	if _, err := an.Call(ctx, &pbReq, &pbBidder); err == nil {
 		t.Fatalf("Should not have gotten an error for missing/invalid size: %v", err)
 	}
 }
 
-func TestIxBasicResponse(t *testing.T) {
+func TestIxInvalidCallEmptyBidIDResponse(t *testing.T) {
+
+	server := httptest.NewServer(
+		http.HandlerFunc(dummyIXServer),
+	)
+	defer server.Close()
+
+	conf := *adapters.DefaultHTTPAdapterConfig
+	an := NewIxAdapter(&conf, server.URL)
+	ctx := context.TODO()
```

Comment:
It's typically used for static analysis, although it's also used when which context to be used is unclear or the function itself does not accept a context yet. `context.Background()` is used by the main goroutine (or parent to the calling function). Both are just empty contexts so I would say both are acceptable in this case since it's a test.

https://golang.org/src/context/context.go?s=7708:7727#L205

Reply:
Thanks for the feedback... always good to learn things.

Reply:
In light of that comment, I think we can leave this as is.  What do you think?

Reply:
yeah, leaving it is fine. I trust Zach's judgment here.